packages/js/src/session/types.ts

```diff
@@ -1,7 +1,11 @@
 import { Subscriber } from '../types';
 
-export type InitializeSessionArgs = {
-  applicationIdentifier: string;
-  subscriber: Subscriber;
-  subscriberHash?: string;
-};
+export type KeylessInitializeSessionArgs = {};
+
+export type InitializeSessionArgs =
+  | KeylessInitializeSessionArgs
+  | {
+      applicationIdentifier?: string;
```

Comment:
Wouldn't it worked if we do keep the appId and subscriberId as requireed?

Reply:
we can but then we would need to do 

```
    this.#session = new Session(
      options.applicationIdentifier
        ? {
            applicationIdentifier: options.applicationIdentifier,
            subscriberHash: options.subscriberHash,
            subscriber: buildSubscriber(options),
          }
        : {},
```

i pushed this change.